tests/test_threadpoolctl.py

```diff
@@ -36,30 +34,15 @@ def test_threadpool_limits_by_prefix(openblas_present, mkl_present, prefix):
         for module in threadpool_info():
             if should_skip_module(module):
                 continue
-            num_threads, filepath = module["num_threads"], module["filepath"]
             if module["prefix"] == prefix:
-                assert num_threads == 1
-            elif "mkl_rt" in module["prefix"] and prefix == "libiomp":
-                # MKL is impacted by a change in the thread pool of Intel
-                # implementation of the OpenMP runtime:
-                assert num_threads == 1
-            else:
-                assert num_threads == original_num_threads[filepath]
+                assert module["num_threads"] == 1
 
     with threadpool_limits(limits={prefix: 3}):
         for module in threadpool_info():
             if should_skip_module(module):
                 continue
-            num_threads, filepath = module["num_threads"], module["filepath"]
-            expected_num_threads = (3, original_num_threads[filepath])
             if module["prefix"] == prefix:
-                assert num_threads in expected_num_threads
-            elif "mkl_rt" in module["prefix"] and prefix == "libiomp":
-                # MKL is impacted by a change in the thread pool of Intel
-                # implementation of the OpenMP runtime:
-                assert num_threads in expected_num_threads
-            else:
-                assert num_threads == original_num_threads[filepath]
+                assert module["num_threads"] <= 3
```

Comment:
> Why did you remove expected_num_threads = (3, original_num_threads[filepath])?

I find the assert more readable this way and furthermore, it tests we are actually limiting to 3 when `original_num_threads[filepath] > 3` which the previous assertion would not do.

> And why did you remove the check that other are left untouched?

Some BLAS implementations (e.g. MKL) can share their threadpool with an OpenMP implementation (e.g. iomp). It could also be the case for OpenBLAS vs libgomp or libomp...